intermediate-js.md

```diff
@@ -3,37 +3,75 @@
 
 2. What is the difference between a div and a span?
+A div tag is used to separate different pieces of the DOM, while span is used to "span" across the entire page of the DOM. div tags will naturally "span" across the entire DOM but are generally styled with a linked CSS file in order to only "contain" a portion of the page. span will always stretch across the entire DOM.
+
+Google:
+I wasn't even close! According to google, span is in-line usually used for a small chunk of HTML. In contrast, a div is block-line (which is equivalent to having a line-break before and after using it) and used to group larger chunks of code.
 
 3. What is a CSS class? When should you use an id instead of a class?
+A CSS class is signified with dot notation. For example, .button would call upon a class named "button" in the linked HTML file and you would they be able to style accordingly. An id on the other hand is signified by a pound sign (or hash tag). An example of this would be, #startButton.
+
+An id is useful when your intention is to style a specific piece of the DOM, and not every element that's of the same type. Extending off the example I used above, if I wanted to turn all my buttons blue, I could give them all a class of .button, but if I wanted one of these buttons a slightly different hue or change the saturation, I could use id in order to individually style it.
 
 4. Name 4 semantic HTML tags.
+I don't know.
+
+Google:
+According to Google, a semantic HTML tag is one that introduces meaning to the web page rather than just presentation. These tags include, but are not limited to, the header tags (<h1> - <h6>), <blockquote>, <code> and <em>.
 
 5. What are three options for creating responsive design?
+I am not familiar with what a responsive design entails, but it sounds like CSS event listeners. A few options of CSS event listeners would be a hover (which applies a style change when hovering over the intended element), a focus (which applies a style change when the element is "in focus") and afterclick (which applies a style after the element has been clicked).
+
+Google:
+According to Google (specifically w3schools), creating a responsive design is about using HTML and CSS to automatically resize, hide, shrink, or enlarge a website, to make it look good on all devices. One option is setting the viewport, which gives the browser instructions on how to control the page's dimensions and scaling. A second option is by creating responsive images by playing around with height and width properties and using percentages rather than static dimensions. The third option is by creating responsive text using something called vw (or viewport width), which allows the font size to change according to the size of the browser window.
```

Comment:
You found some great information! Some of the technologies you can look at that help you implement responsive sites are 1) media queries 2) frameworks/libraries like Bootstrap.